### ISSUE TITLE
Automatic scaling to fit of specific generated image does not preserve aspect ratio

### ISSUE THREAD

**Issue by confluence** [CARTAvis/carta-frontend#1418]
In the latest development version, the example image (`default.fits`) is distorted when it is scaled to fit the view area when it is opened. This is an image with minimal headers. Other images with minimal headers (e.g. `image-10-10-10.fits`, which is generated with [this script](https://github.com/idia-astro/image-generator)) are not affected. I have tried stripping out several header entries from `default.fits` which are not present in `image-10-10-10.fits` and which were generating casacore warnings because they were integer values and not doubles, but the resulting image (`default_minimal.fits`) is also distorted. I am attaching [a zip](https://github.com/CARTAvis/carta-frontend/files/6357870/examples.zip) of all three images.

![Screenshot from 2021-04-22 02-25-26](https://user-images.githubusercontent.com/651030/115707532-33c24880-a36f-11eb-890a-5ef8afe35d6e.png)

**Comment by kswang1029** [comment]
Do you see a pixel is rendered as a rectangle and if you change the viewer aspect ratio do you see the aspect ratio of the rectangle pixel changes too?

**Comment by confluence** [comment]
Yes. Below are screenshots of the pixels after opening and after resizing the viewer.

![Screenshot from 2021-04-22 13-41-53](https://user-images.githubusercontent.com/651030/115708797-b7c90000-a370-11eb-8d52-c73eddc7c1c7.png)

![Screenshot from 2021-04-22 13-42-25](https://user-images.githubusercontent.com/651030/115708818-be577780-a370-11eb-8cb0-4a777b1083fb.png)

**Comment by kswang1029** [comment]
That’s due to the current way we handle pv images. @veggiesaurus help🙈 Somehow the header fits the criteria we consider as pv (or none square pixel image)

**Comment by confluence** [comment]
To be clear, I realise that the correct thing to do may be to add a missing header value to the default image -- I just want to know what that is! It's not obvious from the comparison to the other minimal image.

**Comment by kswang1029** [comment]
Try add cunit1 and cunit2 to be identical, and make sure CDELT1 == -CDELT2

**Comment by confluence** [comment]
But none of these are present in the working image's header.

**Comment by confluence** [comment]
Could the different BITPIX be affecting the result?

**Comment by kswang1029** [comment]
> Could the different BITPIX be affecting the result?

That should be nothing to do with what we check for pv to enable flexible pixel aspect ratio.

**Comment by kswang1029** [comment]
@confluence the reason is casacore "enhanced" the header of image-10-10-10.fits so there are more header entries including CUNIT and CDELT. The other two images after casacore parsing, no CUNIT and CDELT are created so they hit the criteria of rendering them with dynamic pixel aspect ratio. So to fix this, please add CDELT and CUNIT.

**Comment by confluence** [comment]
Oh, I think I have some more information. I made a 1024x1024 image with the generator script to make the header comparison easier, and it turns out that any 2D image generated this way is affected. E.g. [10-10.zip](https://github.com/CARTAvis/carta-frontend/files/6358065/10-10.zip). So the other minimal images appear to be working because they're > 2D.

**Comment by kswang1029** [comment]
after casacore parsing: 
https://user-images.githubusercontent.com/20819712/115712332-3b054880-a3a7-11eb-93f8-0c56f982ff97.mp4

**Comment by confluence** [comment]
I guess casacore makes different guesses for 3D images. I will try fixing the image. But casacore's guessed values for CUNIT* in the 3D file are all blank, and CDELT1 and CDELT2 are both 1.

**Comment by kswang1029** [comment]
The criteria of enabling dynamic aspect ratio should be
if CUNIT1/2 are different, enable it
if CUNIT1/2 are identical and CDELT1 != -CDELT2, enable it

**Comment by confluence** [comment]
But in the 3D file CUNIT1/2 are identical (empty) and CDELT1 != -CDELT2 (they're identical).

There is no way for me to add empty CUNIT1/2 to the headers; when I add literal empty entries they are stripped out by casacore and don't appear in the generated headers.

I can confirm that if I create a 3D default file (from a RGB image) casacore autogenerates the same headers as for the other 3D files. I can use this as a workaround for the default image.

**Comment by kswang1029** [comment]
Make a 3D cube for a nice movie then😆

**Comment by confluence** [comment]
It was 3D originally, but I made it 2D so that it would be smaller. It's so tiny that it doesn't really matter, though!

**Comment by kswang1029** [comment]
@confluence can we close this?
